diff --git a/Makefile.shared b/Makefile.shared
index e753f44..5a5e122 100644
--- a/Makefile.shared
+++ b/Makefile.shared
@@ -151,7 +151,7 @@ DO_GNU_SO=$(CALC_VERSIONS); \
 	SHLIB_SUFFIX=; \
 	ALLSYMSFLAGS='-Wl,--whole-archive'; \
 	NOALLSYMSFLAGS='-Wl,--no-whole-archive'; \
-	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-Bsymbolic -Wl,-soname=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX"
+	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-Bsymbolic -Wl,--default-symver,--version-script=version.map -Wl,-soname=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX"
 
 DO_GNU_APP=LDFLAGS="$(CFLAGS) -Wl,-rpath,$(LIBRPATH)"
 
diff --git a/crypto/cversion.c b/crypto/cversion.c
index ea9f25f..0c5a656 100644
--- a/crypto/cversion.c
+++ b/crypto/cversion.c
@@ -62,7 +62,7 @@
 #include "buildinf.h"
 #endif
 
-const char *SSLeay_version(int t)
+const char *_current_SSLeay_version(int t)
 	{
 	if (t == SSLEAY_VERSION)
 		return OPENSSL_VERSION_TEXT;
@@ -110,8 +110,25 @@ const char *SSLeay_version(int t)
 	return("not available");
 	}
 
-unsigned long SSLeay(void)
+const char *_original_SSLeay_version(int t)
+	{
+	if (t == SSLEAY_VERSION)
+		return "OpenSSL 1.0.0-fips 29 Mar 2010";
+	else
+		return _current_SSLeay_version(t);
+	}
+
+unsigned long _original_SSLeay(void)
+	{
+	return(0x10000003);
+	}
+
+unsigned long _current_SSLeay(void)
 	{
 	return(SSLEAY_VERSION_NUMBER);
 	}
 
+__asm__(".symver _original_SSLeay,SSLeay@");
+__asm__(".symver _original_SSLeay_version,SSLeay_version@");
+__asm__(".symver _current_SSLeay,SSLeay@@OPENSSL_1.0.1");
+__asm__(".symver _current_SSLeay_version,SSLeay_version@@OPENSSL_1.0.1");
diff --git a/version.map b/version.map
new file mode 100644
index 0000000..36038d2
--- /dev/null
+++ b/version.map
@@ -0,0 +1,8 @@
+OPENSSL_1.0.1 {
+    global:
+	    SSLeay;
+	    SSLeay_version;
+    local:
+	    _original*;
+	    _current*;
+};
