diff --git a/autodeps/linux.req b/autodeps/linux.req
index b9a8f99..925d32f 100644
--- a/autodeps/linux.req
+++ b/autodeps/linux.req
@@ -74,7 +74,7 @@ fi
 # --- Script interpreters.
 for f in $scriptlist; do
     [ -r $f -a -x $f ] || continue
-    interp=`head -n 1 $f | sed -e 's/^\#\![ 	]*//' | cut -d" " -f1`
+    interp=`head -n 1 $f | awk '/^#![\s]*/ { sub(/^#![\s]*/, ""); print $1 }'`
     interplist="$interplist $interp"
     case $interp in
     */perl)	perllist="$perllist $f" ;;
