From 2c26c01fe285a56e25b0eb8a0b004e6a0aef2237 Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <davidlt@cern.ch>
Date: Sat, 19 Oct 2013 17:36:04 +0200
Subject: [PATCH 7/8] Fix Darwin requires script (Argument list too long)

Signed-off-by: David Abdurachmanov <davidlt@cern.ch>
---
 autodeps/darwin.req | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/autodeps/darwin.req b/autodeps/darwin.req
index e571710..ec9e6de 100644
--- a/autodeps/darwin.req
+++ b/autodeps/darwin.req
@@ -5,8 +5,8 @@
 ulimit -c 0
 
 filelist=`sed "s/['\"]/\\\&/g"`
-exelist=`echo $filelist | xargs file | grep -F Mach-O | cut -d: -f1 `
-scriptlist=`echo $filelist | xargs file | grep -E ":.* (commands|script) " | cut -d: -f1 `
+exelist=`echo $filelist | xargs -n 100 file | grep -F Mach-O | cut -d: -f1 `
+scriptlist=`echo $filelist | xargs -n 100 file | grep -E ":.* (commands|script) " | cut -d: -f1 `
 
 for f in $exelist; do
     if [ -x $f ]; then
-- 
1.8.0

