From 130813e9fec23637a9f3c9b517745c317e8def55 Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <davidlt@cern.ch>
Date: Sat, 19 Oct 2013 17:32:08 +0200
Subject: [PATCH 6/8] Remove chroot checks

Signed-off-by: David Abdurachmanov <davidlt@cern.ch>
---
 lib/rpmchroot.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/lib/rpmchroot.c b/lib/rpmchroot.c
index 81bb5e5..d1b0d05 100644
--- a/lib/rpmchroot.c
+++ b/lib/rpmchroot.c
@@ -66,9 +66,6 @@ int rpmChrootIn(void)
     } else if (rootState.chrootDone == 0) {
 	if (chdir("/") == 0 && chroot(rootState.rootDir) == 0) {
 	    rootState.chrootDone = 1;
-	} else {
-	    rpmlog(RPMLOG_ERR, _("Unable to change root directory: %m\n"));
-	    rc = -1;
 	}
     }
     return rc;
@@ -91,9 +88,6 @@ int rpmChrootOut(void)
     } else if (rootState.chrootDone == 1) {
 	if (chroot(".") == 0 && fchdir(rootState.cwd) == 0) {
 	    rootState.chrootDone = 0;
-	} else {
-	    rpmlog(RPMLOG_ERR, _("Unable to restore root directory: %m\n"));
-	    rc = -1;
 	}
     }
     return rc;
-- 
1.8.0

